### PR TITLE
Add markdownlint to pre-commit

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Add [markdownlint](https://github.com/DavidAnson/markdownlint) check to pre-commit

Fixed: 

- Incorrect links
- Ordered lists
- Blank lines before and after headers and code blocks
- Types for code blocks
- Trailing spaces
- Spellcheck
- Removed module-timm_custom_modules from api doc

### CODE REVIEW

**Review by alexsu52 — APPROVED** [review]
Thanks for the contribution!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,20 +1,24 @@
...
 
 The latter forms are accepted as pull requests from your own forks of the NNCF repository.
 
 Any contributions must not violate the repository's [LICENSE](./LICENSE) requirements.
 
 ## Installation
-### (Experimental) ONNXRuntime-OpenVINO
+
+## (Experimental) ONNXRuntime-OpenVINO
+
```
Comment (nikita-savelyevv) [inline CONTRIBUTING.md:19]: Shouldn't it be `### (Experimental) ONNXRuntime-OpenVINO`? Otherwise there two same level header one after another

Comment (AlexanderDokuchaev) [reply CONTRIBUTING.md]: Interesting but looks like it should not be here...  was added in https://github.com/openvinotoolkit/nncf/pull/1202

@alexsu52 @vshampor what do you think? Can i remove onnx instruction from CONTRIBUTING.md?
https://github.com/openvinotoolkit/nncf/pull/1996/commits/bbdefae7b90d5c88ed54df13cb9c47e1ff7b88e4

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

**Review by nikita-savelyevv — APPROVED** [review]

**Comment by AlexanderDokuchaev** [comment]
run pytorch pre-commit tests

File: CONTRIBUTING.md (hunk 1 above)
Comment (kshpv) [reply CONTRIBUTING.md]: Looks like we can remove it

**Review by kshpv — APPROVED** [review]

**Pull request merged**